From 47401a0efb45f083ecc726d63bb185d21dc10b06 Mon Sep 17 00:00:00 2001
From: Jon Nettleton <jon.nettleton@gmail.com>
Date: Sun, 24 Aug 2014 09:16:35 +0200
Subject: [PATCH] video: mxc_hdmi: Move update_hdmi_config

Move this to after the read_edid function has completed so we
can make sure and pass the correct information over to the
sound driver.
---
 drivers/video/mxc/mxc_hdmi.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/video/mxc/mxc_hdmi.c b/drivers/video/mxc/mxc_hdmi.c
index b3ea89c3..0aac1e0 100644
--- a/drivers/video/mxc/mxc_hdmi.c
+++ b/drivers/video/mxc/mxc_hdmi.c
@@ -1698,9 +1698,6 @@ static int mxc_hdmi_read_edid(struct mxc_hdmi *hdmi)
 
 	}
 
-	/* Save edid cfg for audio driver */
-	hdmi_set_edid_cfg(hdmi->edid_status, &hdmi->edid_cfg);
-
 	if (ret < 0)
 		return HDMI_EDID_FAIL;
 
@@ -2012,6 +2009,9 @@ static void mxc_hdmi_cable_connected(struct mxc_hdmi *hdmi)
 		break;
 	}
 
+	/* Save edid cfg for audio driver */
+	hdmi_set_edid_cfg(hdmi->edid_status, &hdmi->edid_cfg);
+
 	/* Setting video mode */
 	mxc_hdmi_set_mode(hdmi);
 
-- 
1.7.9.5

